### PR TITLE
doc/dev: add continuous-integration

### PR DESCRIPTION
**Pull request by tchaikov**
for noting down the architecture of our CI. and how we prepare
some of the build dependencies.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
https://ceph--41718.org.readthedocs.build/en/41718/dev/continuous-integration/

File: doc/dev/continuous-integration.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+``make check``, the building hosts in our CI just pull them from our internal
+repos hosting these packages instead of building them.
+
+So far, following packages are prebuilt on ubuntu focal:
+
+libboost
+    packages `boost`_. The packages' names are changed from ``libboost-*`` to
+    ``ceph-libboost-*``, and they are instead installed into ``/opt/ceph``, so
+    they don't interfere with the official ``libboost`` packages shipped by
+    distro. Its build scripts are hosted at https://github.com/tchaikov/ceph-boost.
```
Comment (jdurgin) [inline doc/dev/continuous-integration.rst:127]: perhaps we should move this to the ceph org

**Review by jdurgin — APPROVED** [review]
great to have this written down!

File: doc/dev/continuous-integration.rst (hunk 1 above)
Comment (tchaikov) [reply doc/dev/continuous-integration.rst:127]: on my todo list.

File: doc/dev/continuous-integration.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+   #. set necessary environment variables
+   #. run ``run-make-check.sh``
+
+Chacra
+   is a server offering RESTful API allowing the clients to store and
+   retrieve binary packages. It also creates the repo for uploaded
+   packages automatically. Once a certain repo is created on chacra, the
+   configured shaman server is updated as well, then we can query shaman
+   for the corresponding repo address. Chacra not only hosts ceph packages,
+   it also hosts quite
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply doc/dev/continuous-integration.rst]: fixed and repushed.

File: doc/dev/continuous-integration.rst [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+     {
+       rank=same;
+       package_repos [ peripheries=3 ];
+       pypi;
+       npm;
+     }
+     github:webhooks -> jenkins_controller [ label = "notify", color = "crimson" ];
+     jenkins_controller -> jenkins_agents [ label = "schedule jobs" ];
+     jenkins_agents -> github:git [ label = "git pull" ];
+     jenkins_agents -> shaman [ label = "query for chara repo URL" ];
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply doc/dev/continuous-integration.rst]: fixed.

File: doc/dev/continuous-integration.rst (hunk 1 above)
Comment (tchaikov) [reply doc/dev/continuous-integration.rst:127]: #42131

**Pull request merged**